sapi/fpm/fpm/fpm_main.c

```diff
@@ -1911,6 +1911,12 @@ consult the installation file that came with this distribution, or visit \n\
 			/* check if request_method has been sent.
 			 * if not, it's certainly not an HTTP over fcgi request */
 			if (UNEXPECTED(!SG(request_info).request_method)) {
+				zend_try {
+					zlog(ZLOG_ERROR, "SCRIPT_FILENAME not found in cgi env");
+					SG(sapi_headers).http_response_code = 500;
```

Comment:
Actually I got confused by the message which seems wrong to me. This is about missing `REQUEST_METHOD`. I guess it should be probably 400 in that case.

Reply:
The reason for this message that I didn't get the pipeline from the source, how the method is going to be empty at this point. But the root cause of it is the lack of "SCRIPT_FILENAME". So, from my perspective, it's misconfigured web server.

There is example of fastcgi_param which make a problem behavior on any request
```
fastcgi_param   QUERY_STRING        $query_string;
fastcgi_param   REQUEST_METHOD      $request_method;
fastcgi_param   CONTENT_TYPE        $content_type;
fastcgi_param   CONTENT_LENGTH      $content_length;

fastcgi_param   SCRIPT_NAME         $fastcgi_script_name;
fastcgi_param   REQUEST_URI         $request_uri;
fastcgi_param   DOCUMENT_URI        $document_uri;
fastcgi_param   DOCUMENT_ROOT       $document_root;
fastcgi_param   SERVER_PROTOCOL     $server_protocol;
fastcgi_param   REQUEST_SCHEME      $scheme;
fastcgi_param   HTTPS               $https if_not_empty;

fastcgi_param   GATEWAY_INTERFACE   CGI/1.1;
fastcgi_param   SERVER_SOFTWARE     nginx/$nginx_version;

fastcgi_param   REMOTE_ADDR         $remote_addr;
fastcgi_param   REMOTE_PORT         $remote_port;
fastcgi_param   SERVER_ADDR         $server_addr;
fastcgi_param   SERVER_PORT         $server_port;
fastcgi_param   SERVER_NAME         $server_name;

# PHP only, required if PHP was built with --enable-force-cgi-redirect
fastcgi_param   REDIRECT_STATUS     200;
```

and additional string `fastcgi_param  SCRIPT_FILENAME    $document_root$fastcgi_script_name;` always solves problem.
As mentioned in bug tickets, without this patch fpm answers with empty `HTTP/1.1 200 OK`

If you could point the code, where REQUEST_METHOD left blank - it would be better to patch that place. It should be somewhere related to SCRIPT_FILENAME

Reply:
I don't think this is related to fpm as from what I see, it's just missing `REQUEST_METHOD` fcgi env as set in https://github.com/cfc4n/php-src/blob/8f206845676f93c45ede67f05a79235d2d6dd2b3/sapi/fpm/fpm/fpm_main.c#L1325 . But I would have to try it and have a test for it to be sure.

Btw. there is no requirement that we have to return error in here but I think it makes sense. However not sure if it should go to 7.1 - thinking more about master only...

Reply:
I mean the fact that setting SCRIPT_FILENAME fixes the issue is not related to fpm but might be because the nginx sets it. But as I said not sure and don't have time to look into it now. The main priority is to have test env ready first and then might take a look.